src/managers/commands-parser.js

```diff
@@ -137,32 +137,24 @@ class CommandsParser {
    * @param  {BaseMessage}          message the incoming Discord message
    * @return {Promise<Boolean>}                true if a command was executed successfully, false otherwise
    */
-  async parseDiscordCommand(message) {
+  async processMessage(message) {
+
     const currentPrefix = await this.context.dbManager.getSetting(
       message.source.name,
       message.teamId,
       ServerSettingsTable.SERVER_SETTINGS.commandPrefix.name,
       message.source.DEFAULT_COMMAND_PREFIX
     );
 
-    const currentLocale = await this.context.dbManager.getSetting(
-      message.source.name,
-      message.teamId,
-      ServerSettingsTable.SERVER_SETTINGS.localeName.name
-    );
-
-    const currentUserLocale = await this.context.dbManager.getUserSetting(
-      message.source.name,
-      message.teamId,
-      message.userId,
-      UserSettingsTable.USER_SETTINGS.localeName.name
-    );
+    const commandLangManager = await this.getCommandLangManager(message);
+    const command = this.parseMessage(message, currentPrefix, commandLangManager);
+    if (command) {
+      return await this.executeCommand(message, command, commandLangManager);
```

Comment:
This function must return whether a command was found or not. Now you return nothing, basically, so when this function finishes, the Bot has no chance to understand whether it's a command or just a regular text message and falls into line 108 in index.js file ("c.messageModerator.premoderateDiscordMessage(message);").

Because of that - it will try to premoderate the message even though it was a command to the Bot. For example, if you try to add some bad words while censoring is enabled, it immediately will remove your command message.

Basically, I think that Bot commands should avoid premoderation, so it's better to return something like (command !== null) from here.